File: public/dsa/searching/insert_position_in_sorted_array/naive/insert_position_in_sorted_array.cpp
#include <bits/stdc++.h>
using namespace std;

int getInsertPosition(vector<int> &arr, int key) {
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] >= key) {
            return i;
        }
    }
    return -1;
}

// tests-start
int main() {
    vector<int> arr = {1, 2, 3, 5};
    int key = 4;
    cout << getInsertPosition(arr, key) << endl;
    return 0;
}
// tests-end

/*output
3
*/

File: public/dsa/searching/insert_position_in_sorted_array/naive/insert_position_in_sorted_array.py


from typing import List


class Solution:
    def getInsertPosition(self, arr: List[int], key: int) -> int:
        for i in range(len(arr)):
            if arr[i] >= key:
                return i


# tests-start
arr = [1, 2, 3, 5]
key = 4
sol = Solution()
print(sol.getInsertPosition(arr, key))
# tests-end


'''output
3
'''
